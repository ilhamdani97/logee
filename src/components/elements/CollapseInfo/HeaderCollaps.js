import React from 'react';
import styles from './styles.scoped.css';
import PropTypes from 'prop-types';
import { Typography } from 'logeeport-ui';

const HeaderCollaps = ({ data }) => {
  return (
    <div>
      <Typography className={styles.colapHead} type="p">
        {data.title}
      </Typography>
      <Typography className={styles.colapSubHead} type="p">
        {data.subTitle}
      </Typography>
    </div>
  );
};

HeaderCollaps.defaultProps = {
  data:{},
};

HeaderCollaps.propTypes = {
  data:PropTypes.object,
};


export default HeaderCollaps;
