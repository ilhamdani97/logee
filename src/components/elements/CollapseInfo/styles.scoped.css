.root {
  border: 1px solid #e0e0e0;
  border-radius: 5px;
  margin-bottom: 20px;
}

.heading div:first-child {
  background: #ffffff;
}

.totalHarga {
  text-align: right;
}
.footer {
  background-color: #f5f5f5;
}

.subCaption {
  font-size: 16px;
  color: #000000;
  font-weight: 600;
}

.colapHead {
  font-size: 18px;
  color: #000000;
  font-weight: 600;
}

.colapSubHead {
  margin-top: 4px;
  color: var(--grey-90);
  font-weight: initial;
  font-size: 14px;
  text-transform: initial;
}
