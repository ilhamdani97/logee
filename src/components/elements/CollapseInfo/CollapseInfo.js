import React from 'react';
import styles from './styles.scoped.css';
import PropTypes from 'prop-types';
import { Col, CollapseCard, Row, Typography } from 'logeeport-ui';
import HeaderCollaps from './HeaderCollaps';

const CollapseInfo = ({ children, header, collaps }) => {
  return (
    <div className={styles.root}>
      <CollapseCard
        className={styles.heading}
        collapsable = {collaps}
        collapseDefault={true}
        title={<HeaderCollaps data={header}/>}
      >
        {children}
      </CollapseCard>
      <Row className={styles.footer} justify="space-between">
        <Col xs="4">
          <Typography className={styles.subCaption}>
            Total
          </Typography>
        </Col>
        <Col className={styles.totalHarga} xs="4">{header.total}</Col>
      </Row>
    </div>
  );
};

CollapseInfo.defaultProps = {
  children: null,
  className: '',
  collaps: false,
  header:{},
};

CollapseInfo.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  collaps: PropTypes.bool,
  header:PropTypes.object,
};

export default CollapseInfo;



