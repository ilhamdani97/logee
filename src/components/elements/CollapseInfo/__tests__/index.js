import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import CollapseInfo from '../CollapseInfo';

const header =  {
  'title': 'Armada',
  'subTitle': 'Total armada berdasarkan jenis dan kapasitasnya pada periode waktu yang dipilih.',
  'total': '24'
};
describe('src/components/elements/CollapseInfo', () => {

  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<CollapseInfo header={header}/>);
    expect(tree).toMatchSnapshot();
  });
});
