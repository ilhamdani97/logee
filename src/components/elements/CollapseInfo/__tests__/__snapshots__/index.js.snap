// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/elements/CollapseInfo render 1`] = `
<div>
  <CollapseCard
    className=""
    collapsable={true}
    collapseDefault={true}
    icon={null}
    iconBgColor=""
    title={
      <HeaderCollaps
        data={
          Object {
            "subTitle": "Total armada berdasarkan jenis dan kapasitasnya pada periode waktu yang dipilih.",
            "title": "Armada",
            "total": "24",
          }
        }
      />
    }
  />
  <Row
    alignContent=""
    alignItem=""
    className=""
    direction=""
    id=""
    justify="space-between"
    noGutter={false}
  >
    <Col
      className=""
      id=""
      lg=""
      md=""
      offsetLg=""
      offsetMd=""
      offsetSm=""
      offsetXL=""
      offsetXs=""
      orderLg=""
      orderMd=""
      orderSm=""
      orderXl=""
      orderXs=""
      sm=""
      xl=""
      xs="4"
    >
      <Typography
        spacing=""
      >
        Total
      </Typography>
    </Col>
    <Col
      className=""
      id=""
      lg=""
      md=""
      offsetLg=""
      offsetMd=""
      offsetSm=""
      offsetXL=""
      offsetXs=""
      orderLg=""
      orderMd=""
      orderSm=""
      orderXl=""
      orderXs=""
      sm=""
      xl=""
      xs="4"
    >
      24
    </Col>
  </Row>
</div>
`;
